app/lib/zif/services/input_service.rb

```diff
@@ -123,6 +139,24 @@ def process_click
       end
       # rubocop:enable Metrics/PerceivedComplexity
 
+      # @api private
+      def process_key_event
+        return if @key_pressables.empty?
+
+        keys_down = $gtk.args.inputs.text
+        $gtk.args.inputs.keyboard.key_down.truthy_keys.each do |k|
+          keys_down << k if Zif::UI::Input::DEFAULT_SPECIAL_KEYS.include?(k)
```

Comment:
this is wrong to be using the default, right now I'm not sure how to figure out which keys are convertable to strings and can be added to the label and which ones are control characters. Might just make another list of control characters... TBD.